src/renderers/dom/client/utils/DOMChildrenOperations.js

```diff
@@ -42,6 +43,47 @@ function insertLazyTreeChildAt(parentNode, childTree, referenceNode) {
   DOMLazyTree.insertTreeBefore(parentNode, childTree, referenceNode);
 }
 
+function isClosingTextComment(node) {
+  return node.nodeType === 8 && node.nodeValue === ' /react-text ';
+}
+
+function removeDelimitedText(startNode) {
+  var parentNode = startNode.parentNode;
+  while (true) {
+    var node = startNode.nextSibling;
+    if (isClosingTextComment(node)) {
+      return node;
+    } else {
+      parentNode.removeChild(node);
+    }
+  }
+}
+
+function replaceDelimitedText(openingComment, stringText) {
+  var parentNode = openingComment.parentNode;
+  var nodeAfterComment = openingComment.nextSibling;
+  if (isClosingTextComment(nodeAfterComment)) {
+    // There are no text nodes between the opening and closing comments; insert
+    // a new one if stringText isn't empty.
+    if (stringText) {
```

Comment:
I made it so that if it's an empty string it doesn't insert an empty text node between the comments. Not really sure which is preferable.